packages/expo/src/Expo.fx.tsx

```diff
@@ -82,4 +83,11 @@ if (__DEV__) {
       originalSetWrapperComponentProvider(() => PatchedProviderComponent);
     };
   }
+
+  const proxiedNativeModules = createProxyForNativeModules(ReactNative.NativeModules);
+  Object.defineProperty(ReactNative, 'NativeModules', {
+    get() {
+      return proxiedNativeModules;
+    },
```

Comment:
Preserve extensibility:
```suggestion
    get() {
      return proxiedNativeModules;
    },
    configurable: true,
    enumerable: true,
```